src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs

```diff
@@ -518,7 +517,7 @@ internal static void EnsureGroupPreservationColumn(IHostEnvironment env, ref IDa
                     // instead of having two hash transformations.
                     var origStratCol = samplingKeyColumn;
                     samplingKeyColumn = data.Schema.GetTempColumnName(samplingKeyColumn);
-                    var columnOptions = new HashingEstimator.ColumnOptionsInternal(samplingKeyColumn, origStratCol, 30, (uint)(seed ?? host.Rand.Next()));
+                    var columnOptions = new HashingEstimator.ColumnOptionsInternal(samplingKeyColumn, origStratCol, 30, (uint)(seed ?? env.Seed));
```

Comment:
>env.Seed [](start = 135, length = 8)

This can also be null, can't it?

Reply:
Yes, if `env.Seed` is also null, it will fall back to the `HashingEstimator` default seed:
https://github.com/dotnet/machinelearning/blob/46e7dc6269ef4a55c0e5788f0c327ef2029ee2f2/src/Microsoft.ML.Data/Transforms/Hashing.cs#L1129-L1132

Reply:
But you are casting it to `uint` here. If it's null it will throw an exception.

---
In reply to: [375521058](https://github.com/dotnet/machinelearning/pull/4775#discussion_r375521058) [](ancestors = 375521058)

Reply:
Ah yes, you're right. Thanks for catching that @yaeldekel. I've made it nullable like in `GenerateNumberTransform` above this.